src/AnalogSensor.cpp

```diff
@@ -9,7 +9,12 @@ AnalogSensor::AnalogSensor(const uint32_t id, const bool criticality, const uint
     this->pin = pin;
 }
 
-bool AnalogSensor::healthCheck() const { return analogRead(pin) == 0; }
+Health AnalogSensor::healthCheck() const
+{
+    if (analogRead(pin) == 0) { return HEALTHY; }
+    if (criticality) { return CRITICAL; }
+    return UNRESPONSIVE;
```

Comment:
I think this is backwards, if an analogPin reads 0 then it is unplugged and Unresponsive

Reply:
I switched it to be !=. Still not the best implementation but we will wait until the new guy tells us what he's got.